airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py

```diff
@@ -101,58 +83,15 @@ def get_authenticator(config):
             refresh_token=config.get("refresh_token"),
         )
 
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
-        config = self._validate_and_transform(config)
-        authenticator = self.get_authenticator(config)
-        url = f"{PinterestStream.url_base}user_account"
-        try:
-            auth_headers = {"Accept": "application/json", **authenticator.get_auth_header()}
-            session = requests.get(url, headers=auth_headers)
-            session.raise_for_status()
-            return True, None
-        except requests.exceptions.HTTPError as e:
-            if "401 Client Error: Unauthorized for url" in str(e):
-                return False, f"Try to re-authenticate because current refresh token is not valid. {e}"
-            else:
-                return False, e
-        except requests.exceptions.RequestException as e:
-            return False, e
-
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         config["authenticator"] = self.get_authenticator(config)
         report_config = self._validate_and_transform(config, amount_of_days_allowed_for_lookup=913)
-        config = self._validate_and_transform(config)
-        status = ",".join(config.get("status")) if config.get("status") else None
-
-        ad_accounts = AdAccounts(config)
-        ads = Ads(ad_accounts, config=config, status_filter=status)
-        ad_groups = AdGroups(ad_accounts, config=config, status_filter=status)
-        campaigns = Campaigns(ad_accounts, config=config, status_filter=status)
-        boards = Boards(config)
-        board_sections = BoardSections(boards, config=config)
-        return [
-            ad_accounts,
-            AdAccountAnalytics(ad_accounts, config=config),
-            ads,
-            AdAnalytics(ads, config=config),
-            ad_groups,
-            AdGroupAnalytics(ad_groups, config=config),
-            boards,
-            BoardPins(boards, config=config),
-            board_sections,
-            BoardSectionPins(board_sections, config=config),
-            campaigns,
-            CampaignAnalytics(campaigns, config=config),
+
+        declarative_streams = super().streams(config)
+        ad_accounts = [stream for stream in declarative_streams if stream.name == "ad_accounts"][0]
+        report_streams = [
```

Comment:
why are we unable to convert Pinterest report streams into low-code streams?

Reply:
As I understood, these streams involve async data fetch (like init report -> check status -> download report). Are we able to handle such behavior in low-code?

Reply:
got it, that makes sense. We don't currently support async polling within the framework right now. If you could just add a comment that report streams use async polling just so it's documented why